### PR TITLE
Kobel kostyantyn

### PR DESCRIPTION
**Pull request by kolyasalubov**

### CODE REVIEW

File: HW1/kkobel/task1.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+name = input('What is your name?: ')
+age =  int(input('How old are you?: '))
+adress = input('Where do you live?: ')
+
+print('Hello,', name.capitalize())
+print('Your age is:', age)
+print('You live in:', adress.capitalize())
```
Comment (kolyasalubov) [inline HW1/kkobel/task1.py:7]: good job

File: HW1/kkobel/task2.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+subtraction = number_a - number_b
+multiplication = number_a * number_b
+division = int(number_a / number_b)
+exponentiation = number_a ** number_b
+
+print('Addition of numbers a + b: ', addition)
+print('Subtraction of numbers a - b: ', subtraction)
+print('Multiplication of numbers a * b: ', multiplication)
+print('Divivsion of numbers a / b: ', division)
+print('Exponentiation of numbers a ** b: ', exponentiation)
```
Comment (kolyasalubov) [inline HW1/kkobel/task2.py:14]: good job

File: HW3/kkobel/task1.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    In the face of ambiguity, refuse the temptation to guess.
+    There should be one-- and preferably only one --obvious way to do it.
+    Although that way may not be obvious at first unless you're Dutch.
+    Now is better than never.
+    Although never is often better than *right* now.
+    If the implementation is hard to explain, it's a bad idea.
+    If the implementation is easy to explain, it may be a good idea.
+    Namespaces are one honking great idea -- let's do more of those!
+'''
+find_better_in_string = set_rules_of_Zen.find('better')
```
Comment (kolyasalubov) [inline HW3/kkobel/task1.py:24]: method find() return index of input parameters,
in this case you need to use 
find_better_in_string = set_rules_of_Zen.count('better')

File: HW3/kkobel/task1.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+find_never_in_string = set_rules_of_Zen.find('never')
```
Comment (kolyasalubov) [inline HW3/kkobel/task1.py:25]: find_never_in_string = set_rules_of_Zen.count('never')

File: HW3/kkobel/task1.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+find_is_in_string = set_rules_of_Zen.find('is')
```
Comment (kolyasalubov) [inline HW3/kkobel/task1.py:26]: find_is_in_string = set_rules_of_Zen.count('is')

File: HW3/kkobel/task2.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    digit_of_number = four_integer_natural_number % 10
+    reversed__of_number = reversed__of_number * 10 + digit_of_number
+    four_integer_natural_number //= 10
+print(f"Reversed number is: {reversed__of_number}")    
+
+# Third part of Task
+
+change_to_list_of_string = str(reversed__of_number)
+sort_list_of_number = sorted(change_to_list_of_string)
+print(f"Sort of numbers is : {sort_list_of_number}")
```
Comment (kolyasalubov) [inline HW3/kkobel/task2.py:27]: good job

File: HW3/kkobel/task3.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+first_string = first_string * second_string 
+second_string = first_string // second_string 
+first_string = first_string // second_string 
+print(f"First string become: {first_string}, Second string become: {second_string}")
+
+# Fiveth of variant
+first_string = first_string ^ second_string 
+second_string = first_string ^ second_string 
+first_string = first_string ^ second_string 
+print(f"First string become: {first_string}, Second string become: {second_string}")
```
Comment (kolyasalubov) [inline HW3/kkobel/task3.py:31]: excellent job

File: HW4/kkobel/factorial.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+elif factorial_of_number < 0:
+    print(f"Factorial {factorial_of_number} isn't positive and can't be exist")
+else:
+    while count < factorial_of_number+1:
+        factorial_n *= count
+        print(factorial_n)
+        count += 1
+        list_of_factorial.append(factorial_n)
+    print(list_of_factorial)
+    print(f"Factorial of number {factorial_of_number} is: {factorial_n}")
```
Comment (kolyasalubov) [inline HW4/kkobel/factorial.py:30]: good job

File: HW5/kkobel/list_of_float_numbers.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
+user_input = int(input("Enter count of numbers: "))
+
+list_of_integer_numbers = [(int(input("Enter your number: "))) for number in range(user_input) ]
+
+print(f"List of integer numbers are: {list_of_integer_numbers}")
+
+list_of_float_numbers = []
+for element in list_of_integer_numbers:
+    list_of_float_numbers.append(float(element))
+
```
Comment (kolyasalubov) [inline HW5/kkobel/list_of_float_numbers.py:10]: good job

File: HW6/kkobel/task1.py [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+    if number %2 == 0:
+        list_numbers_that_divisible_by_2.append(number)    
+    elif number %3 == 0 and number %2 !=0:
+        list_numbers_that_divisible_by_3.append(number)
+    elif number %2 != 0 and number %3 != 0:
+        list_numbers_that__are_not_divisible_by_2_and_3.append(number)    
+
+print(f"Numbers that divisible by 2: {list_numbers_that_divisible_by_2}")
+print(f"Numbers that divisible by 3: {list_numbers_that_divisible_by_3}")
+print(f"Numbers that are not divisible by 2 and 3: {list_numbers_that__are_not_divisible_by_2_and_3}")
```
Comment (kolyasalubov) [inline HW6/kkobel/task1.py:22]: good job

File: HW6/kkobel/user_enter_login_message.py [hunk 11]
```diff
@@ -0,0 +1,10 @@
+user_input_login = input("Enter your login,please: ")
+
+while True:
+    if user_input_login == "First":
+        print(f"Hello {user_input_login}, Great works you are logged in")
+        break
+    else:
+        print(f"You entered {user_input_login}, it's wrong login", end=" " )
+        print(":Try again")
+        user_input_login = input("Enter your login,please: ")
```
Comment (kolyasalubov) [inline HW6/kkobel/user_enter_login_message.py:10]: good job 
consider  another way of solution
user_name = input('Hello, please input your Log in:')

while user_name != 'First':
    # print('Error: wrong username, please try one more time.')
    user_name = input('Error: wrong username, please try one more time. Username:')
    # print(user_name)

else:
    print('Greeting. Access granted!!!', user_name)

File: HW7/kkobel/calculate_distance_between_two_pairs.py [hunk 12]
```diff
@@ -0,0 +1,7 @@
+def distance(x1, y1, x2, y2):
+    distance_of_two_pairs = (((x2-x1) ** 2) + ((y2-y1) ** 2))
+    result_of_sqrt = distance_of_two_pairs ** 0.5
+    result_of_round = float("%0.2f" % result_of_sqrt)
+    return result_of_round
+
+print(f"Distance of two ordered pairs: is: {distance(1, 1, 0, 0)}")
```
Comment (kolyasalubov) [inline HW7/kkobel/calculate_distance_between_two_pairs.py:7]: good job

File: HW7/kkobel/change_str_to_lower_characters.py [hunk 13]
```diff
@@ -0,0 +1,5 @@
+def filter_words(string):
+    change_to_capitalize_str = " ".join(string.split()).capitalize()
+    return change_to_capitalize_str
+
+print(filter_words("CAN    YOU     HEAR     ME"))
```
Comment (kolyasalubov) [inline HW7/kkobel/change_str_to_lower_characters.py:5]: good job

File: HW7/kkobel/convert_num_to_string.py [hunk 14]
```diff
@@ -0,0 +1,3 @@
+def number_to_string(num):
+    convert_num_to_string = str(num)
+    return convert_num_to_string
```
Comment (kolyasalubov) [inline HW7/kkobel/convert_num_to_string.py:3]: good job

File: HW7/kkobel/fixed_error_in_task.py [hunk 15]
```diff
@@ -0,0 +1,5 @@
+def greet(name):
+    return "Hello, my love!" if name == "Johnny" else "Hello, {name}!".format(name=name)  
+   
+print(greet('Johnny'))
+print(greet('Someone of people'))
```
Comment (kolyasalubov) [inline HW7/kkobel/fixed_error_in_task.py:5]: good job

File: HW7/kkobel/reverse_str.py [hunk 16]
```diff
@@ -0,0 +1,8 @@
+def reversing(st: str):
+    result_of_split_and_join = " ".join(st.replace(" ", " ").split(" ")[::-1])
+    return result_of_split_and_join
+
+print(reversing('Hello World'))
+
+print(reversing('Hi There.'))
+print(reversing('I am an expert at this'))
```
Comment (kolyasalubov) [inline HW7/kkobel/reverse_str.py:8]: good job

File: HW7/kkobel/task1.py [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+        return number1
+    elif number2 > number1:
+        return number2
+    elif number1 == number2:
+        print(f"Number1: {number1} is equal Number2: {number2}")
+
+result_of_two_equals_numbers = return_the_largest_number_of_two_numbers(12,12)
+first_result_of_two_numbers = print(f"Number1 is the largest than Number2 and equal:",return_the_largest_number_of_two_numbers(12,10))
+second_result_of_two_numbers = print(f"Number2 is the largest than Number1 and equal:",return_the_largest_number_of_two_numbers(100,1000))
+print(return_the_largest_number_of_two_numbers.__doc__)
```
Comment (kolyasalubov) [inline HW7/kkobel/task1.py:17]: good job

File: HW7/kkobel/task2.py [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+        calculate_area_of_rectangle(int(input("Enter length: ")), int(input("Enter width: ")))
+    elif user_choice == "triangle":
+        calculate_area_of_triangle(int(input("Enter side_a: ")), int(input("Enter height: ")))
+    elif user_choice == "circle":
+        calculate_area_of_circle(int(input("Enter radius: ")))
+    else:
+        print(f"You entered {user_choice}, it is not correct function.Try again")
+        user_choice_input = area_of_figures(input("Choice what you have to calculate(rectangle, triangle,circle): "))                 
+    
+user_choice_input = area_of_figures(input("Choice what Area you have to calculate(rectangle, triangle,circle): "))
```
Comment (kolyasalubov) [inline HW7/kkobel/task2.py:31]: good job

File: HW7/kkobel/task3.py [hunk 19]
```diff
@@ -0,0 +1,12 @@
...
+    Function that calculates the number of characters
+    '''
+    change_to_enumerate = enumerate(string)
+    change_to_dict = dict(change_to_enumerate)
+    list_of_dictionary = {}
+    for val in change_to_dict.values():
+        count_of_value = string.count(val)
+        list_of_dictionary.update({val: count_of_value})
+    print(f"Your count of elements: {list_of_dictionary}") 
+user_input_string = count_numbers_in_string(input("Enter your string: "))
```
Comment (kolyasalubov) [inline HW7/kkobel/task3.py:12]: consider this solution

def count_symbols_in_row(word):
    result = {}
    for item in word:
        if item in result:
            continue
        else:
            result.update({str(item): word.count(item)})
    return result

print(count_symbols_in_row("abracadabra"))
print(count_symbols_in_row("100500"))
print(count_symbols_in_row("##$$aa"))
print(count_symbols_in_row(input("Enter your word")))

File: HW3/kkobel/task1.py (hunk 3 above)
Comment (KostyaKobel) [reply HW3/kkobel/task1.py:24]: Changed method find at count in 24 string

File: HW3/kkobel/task1.py (hunk 4 above)
Comment (KostyaKobel) [reply HW3/kkobel/task1.py:25]: Changed method find at count in 25 string

File: HW3/kkobel/task1.py (hunk 5 above)
Comment (KostyaKobel) [reply HW3/kkobel/task1.py:26]: Changed method find at count in 26 string

File: HW7/kkobel/task3.py (hunk 19 above)
Comment (KostyaKobel) [reply HW7/kkobel/task3.py:12]: Refactor logic of task

File: HW3/kkobel/task1.py (hunk 3 above)
Comment (kolyasalubov) [reply HW3/kkobel/task1.py:24]: well done)!

File: HW3/kkobel/task1.py (hunk 5 above)
Comment (kolyasalubov) [reply HW3/kkobel/task1.py:26]: well done)!

File: HW7/kkobel/task3.py (hunk 19 above)
Comment (kolyasalubov) [reply HW7/kkobel/task3.py:12]: good job

**Pull request merged**